jdaviz/tests/test_data_formats.py

```diff
@@ -6,54 +6,69 @@
 from jdaviz.core.config import list_configurations
 
 
-def create_sdss(n_dim=2):
+def create_sdss(n_dim=2, **kwargs):
     """ create a fake SDSS single fiber spectral fits file """
     primary = fits.PrimaryHDU(header=fits.Header({'TELESCOP': "SDSS 2.5-M", 'FIBERID': 555}))
     ext = fits.BinTableHDU(name='DATA', header=fits.Header({'TTYPE3': "ivar"}))
-    hdulist = fits.HDUList([primary, ext])
-    return hdulist
+    return fits.HDUList([primary, ext])
 
 
-def create_manga(n_dim):
+def create_manga(n_dim=None, **kwargs):
     """ create a fake SDSS MaNGA fits file """
     primary = fits.PrimaryHDU(header=fits.Header({'TELESCOP': "SDSS 2.5-M"}))
     ext = fits.ImageHDU(name='FLUX', header=fits.Header({'INSTRUME': "MaNGA"}))
     shape = (1,) if n_dim == 1 else (1, 2,) if n_dim == 2 else (1, 2, 3) if n_dim == 3 else (1,)
     ext.data = np.empty(shape)
-    hdulist = fits.HDUList([primary, ext])
-    return hdulist
+    return fits.HDUList([primary, ext])
 
 
-def create_jwst(n_dim):
+def _create_bintable(name, ver=1):
+    return fits.BinTableHDU.from_columns(
+                [fits.Column(name='target', format='20A', array=np.ones(3)),
+                 fits.Column(name='V_mag', format='E', array=np.ones(3))],
+                name=name, ver=ver
+            )
+
+
+def create_jwst(n_dim=None, multi=None, ext=None):
     """ create a fake JWST fits file """
     primary = fits.PrimaryHDU(header=fits.Header({'TELESCOP': "JWST"}))
-    ext = fits.ImageHDU(name='ASDF')
+    asdf = fits.ImageHDU(name='ASDF')
```

Comment:
Not sure what this does, but it doesn't seem right to create ASDF without using `asdf` or `gwcs`.

Reply:
This function creates a fake JWST file with mostly empty data extensions, just enough to get the format and structure right so I can test the various data format identification functions.   Since the fake `ASDF` extension isn't really being used, tested or populated, I think it's ok to not make a fully realized one using `asdf` or `gwcs`.   

I think if we wanted to create a more robust fake file generator to be used more generally down the line, then, yeah, we'd certainly want more realistic fake extensions.